### PR TITLE
CalendarFrame Test

### PR DESCRIPTION
**Pull request by jaadams5**
I'm interested in a minor cosmetic change concerning time.  I realized it would have the greatest impact on CalendarFrame and there were not tests for that.  So this is a test for CalendarFrame.  I'm not really interested in the GUI actions so the interesting part of the test is getting the data from the tabs.  That data is derived from "today" so I made a way to pass the date to CalendarFrame so the test was not time dependent.  I did something similar to avoid having the test depend upon a random number.  On the test itself there are some assertions that are probably not useful.  I'm not convinced making a parametrized test of tabs and data but I could cave if someone else feels strongly.

### CODE REVIEW

**Comment by jaadams5** [comment]
So my Swing test needs to not throw java.awt.HeadlessException.  Will investigate.

**Comment by jaadams5** [comment]
#1454 seems to fix the test failure without refactoring the code.

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (midgleyc) [inline test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java:41]: You can get better failure messages using `assertThat`, e.g.

```
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.instanceOf;

assertThat(aTab, instanceOf(RequestPane.class));
```

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (midgleyc) [inline test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java:66]: this is a bit run-on. Can you not strip the HTML, and test directly against it (including the tags)?

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (midgleyc) [inline test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java:36]: may as well `fail()` here, or even just let the error propagate to fail the test.

File: .github/workflows/check.yml
Comment (Veracity0) [inline .github/workflows/check.yml:25]: So this all from Rinn's draft PR, right?

How about if we get that merged first, on its own, rather than pulling it into this PR?

**Review by Veracity0 — DISMISSED** [review]

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java:41]: OK

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java:36]: OK

File: .github/workflows/check.yml
Comment (jaadams5) [reply .github/workflows/check.yml:25]: I merged here to establish that it solved the problem.  I will not push this until Rinn's has been approved and main merged back here.

**Comment by jaadams5** [comment]
Got concurrent mod errors from GearChangePanelTest.  Hypothesis is that the Script menu was getting initialized more than once.  Simulated login and broke up tests.  Need to revise the expected data, possibly make a parameterized test and consider using Cleanups instead of beforeAll/afterAll.

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java]: OK.

**Comment by jaadams5** [comment]
Draft until I figure the errors out.  I think there is an EOL issue on Ubuntu and the concurrent mod errors are because the failed tests aren't cleaning up.

File: test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java
Comment (jaadams5) [reply test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java:79]: Do you know an IntelliJ command that would do that?  spotless didn't.

Comment (jaadams5) [reply test/net/sourceforge/kolmafia/swingui/CalendarFrameTest.java]: nevermind.  figured it out.

File: .github/workflows/check.yml
Comment (jaadams5) [reply .github/workflows/check.yml]: And Rinn's was pushed.  Dismissing review since can't figure out how to resolve.

**Review by midgleyc — APPROVED** [review]
Thanks!

Are those "\s"s both real and intended, though?

**Comment by jaadams5** [comment]
> Thanks!
> 
> Are those "\s"s both real and intended, though?

I copied the string as displayed in the debugger and pasted it into the code.  The calendar does display differently based upon the moon phases so maybe they would make more sense when the moon data was different?

**Comment by midgleyc** [comment]
This change seems to have broken the Jenkins build. I suppose a similar change needs making to the `jenkinsfile` after the server is updated to install `xvfb` as was installed here.

**Pull request merged**
